pyro/distributions/gaussian_scale_mixture.py

```diff
@@ -0,0 +1,152 @@
+from __future__ import absolute_import, division, print_function
+import math
+
+import torch
+from torch.autograd import Function
+from torch.autograd.function import once_differentiable
+from torch.distributions import constraints, Categorical
+
+from pyro.distributions.torch_distribution import TorchDistribution
+from pyro.distributions.util import sum_leftmost
+
+
+class GaussianScaleMixture(TorchDistribution):
+    """
+    Mixture of Normal distributions with zero mean and diagonal covariance
+    matrices.
+
+    That is, this distribution is a mixture with K components, where each
+    component distribution is a D-dimensional Normal distribution with zero
+    mean and a D-dimensional diagonal covariance matrix. The K different
+    covariance matrices are controlled by the parameters `coord_scale` and
+    `component_scale`.  That is, the covariance matrix of the k'th component is
+    given by
+
+    Sigma_ii = (component_scale_k * coord_scale_i) ** 2   (i = 1, ..., D)
+
+    where `component_scale`_k is a positive scale factor and `coord_scale`_i
+    are positive scale parameters shared between all K components. The mixture
+    weights are controlled by a K-dimensional vector of softmax logits,
+    `component_logits`. This distribution implements pathwise derivatives for
+    samples from the distribution.  This distribution does not currently
+    support batched parameters.
+
+    See reference [1] for details on the implementations of the pathwise
+    derivative. Please consider citing this reference if you use the pathwise
+    derivative in your research.
+
+    [1] Pathwise Derivatives for Multivariate Distributions, Martin Jankowiak &
+    Theofanis Karaletsos. arXiv:1806.01856
+
+    Note that this distribution supports both even and odd dimensions, but the
+    former should be more a bit higher precision, since it doesn't use any erfs
+    in the backward call. Also note that this distribution does not support
+    D = 1.
+
+    :param torch.tensor coord_scale: D-dimensional vector of scales
+    :param torch.tensor component_logits: K-dimensional vector of logits
+    :param torch.tensor component_scale: K-dimensional vector of scale multipliers
+    """
+    has_rsample = True
+    arg_constraints = {"component_scale": constraints.positive, "coord_scale": constraints.positive,
+                       "component_logits": constraints.real}
+
+    def __init__(self, coord_scale, component_logits, component_scale):
+        self.dim = coord_scale.size(0)
+	if self.dim < 2:
```

Comment:
What happened to the indentation here?

Reply:
![mug](https://themighty.com/wp-content/uploads/2017/08/ThinkstockPhotos-496036805-640x213.jpg?v=1502444240)